evm/hyperliquid.js

```diff
@@ -61,23 +38,29 @@ function constructPhantomAgent(hash, isMainnet) {
     };
 }
 
-async function signL1Action(wallet, action, activePool, nonce, isMainnet) {
+async function signL1Action(wallet, action, activePool, nonce, isMainnet, chain) {
     const hash = actionHash(action, activePool, nonce);
     const phantomAgent = constructPhantomAgent(hash, isMainnet);
 
-    const signature = await wallet._signTypedData(HYPERLIQUID_CONFIG.domain, { Agent: HYPERLIQUID_CONFIG.types.Agent }, phantomAgent);
+    // Use chain.hypercore.domain if available, otherwise fall back to default
+    const domain = chain?.hypercore?.domain;
+    const agent = [
+        { name: 'source', type: 'string' },
+        { name: 'connectionId', type: 'bytes32' },
+    ];
+
+    const signature = await wallet._signTypedData(domain, { Agent: agent }, phantomAgent);
     const sig = ethers.utils.splitSignature(signature);
 
     return { r: sig.r, s: sig.s, v: sig.v };
 }
 
-async function sendRequest(action, signature, nonce, isMainnet) {
+async function sendRequest(action, signature, nonce, chain) {
     const payload = { action, signature, nonce };
-    const endpoint = isMainnet ? HYPERLIQUID_CONFIG.endpoints.mainnet : HYPERLIQUID_CONFIG.endpoints.testnet;
+    const endpoint = `${chain.hypercore.url}/exchange`;
 
     const curlCommand = `curl -s -X POST "${endpoint}" \
         -H "Content-Type: application/json" \
-        -H "User-Agent: ${HYPERLIQUID_CONFIG.userAgent}" \
         -d '${JSON.stringify(payload)}' \
         --connect-timeout 15 \
         --max-time 30`;
```

Comment:
use `execSync` instead of `await execAsync` maybe?

Reply:
throw an error if stderr is non empty for any reason and not just for curl (or more reliable, check the exit code?)